/*
 * Copyright (c) 2014-2016 - Adjacent Link LLC, Bridgewater, New Jersey
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 *
 *  * Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 *  * Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in
 *    the documentation and/or other materials provided with the
 *    distribution.
 *  * Neither the name of Adjacent Link LLC nor the names of its
 *    contributors may be used to endorse or promote products derived
 *    from this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
 * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
 * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 *
 * See toplevel COPYING for more information.
 */

 /*
  * !!!
  * !!! THIS FILE IS AUTOMATICALLY GENERATED. DO NOT MODIFY
  * !!!
  */

syntax = "proto2";

package OpenTestPoint;

option optimize_for = SPEED;

import "otestpoint/emane/physicallayerreceivepowertable.proto";
import "otestpoint/emane/pathlosseventtable.proto";
import "otestpoint/emane/eventreceptiontable.proto";
import "measurementtable.proto";


message Measurement_emane_emanesh_routing_general
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_emanesh_routing_general"];
    required string module = 2 [default = "otestpoint.emane.emanesh_routing"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

  optional float avgdownstreamprocessingdelay0 = 2 ;
}

message Measurement_emane_emanesh_routing_mac
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_emanesh_routing_mac"];
    required string module = 2 [default = "otestpoint.emane.emanesh_routing"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable0 = 2 ;
   optional MeasurementTable broadcastpacketdroptable0 = 3 ;
   optional MeasurementTable eventreceptiontable = 4 ;
   optional MeasurementTable neighbormetrictable = 5 ;
   optional MeasurementTable neighborstatustable = 6 ;
   optional MeasurementTable unicastpacketaccepttable = 7 ;
   optional MeasurementTable unicastpacketdroptable = 8 ;
}

message Measurement_emane_emanesh_routing_phy
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_emanesh_routing_phy"];
    required string module = 2 [default = "otestpoint.emane.emanesh_routing"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable antennaprofileeventinfotable = 2 ;
   optional MeasurementTable broadcastpacketaccepttable0 = 3 ;
   optional MeasurementTable broadcastpacketdroptable0 = 4 ;
   optional EventReceptionTable eventreceptiontable = 5 ;
   optional MeasurementTable locationeventinfotable = 6 ;
   optional PathlossEventTable pathlosseventtable = 7 ;
   optional PhysicalLayerReceivePowerTable receivepowertable = 8 ;
   optional MeasurementTable unicastpacketaccepttable = 9 ;
   optional MeasurementTable unicastpacketdroptable = 10 ;
}

message Measurement_emane_emanesh_routing_transport
{
  message Description
  {
    required string name = 1 [default = "Measurement_emane_emanesh_routing_transport"];
    required string module = 2 [default = "otestpoint.emane.emanesh_routing"];
    required uint32 version = 3 [default = 1];
  }
  // do not set description
  optional Description description = 1;

   optional MeasurementTable broadcastpacketaccepttable = 2 ;
   optional MeasurementTable broadcastpacketdroptable = 3 ;
   optional MeasurementTable eventreceptiontable = 4 ;
   optional MeasurementTable unicastpacketaccepttable = 5 ;
   optional MeasurementTable unicastpacketdroptable = 6 ;
}
